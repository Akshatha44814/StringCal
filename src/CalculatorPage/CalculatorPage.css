.input {
  padding: 10px;
  width: 360px;
  border: 1px solid #ececec;
  border-radius: 2px;
}
.resultInput {
  margin-bottom: 5%;
}
button[disabled] {
  border: 1px solid #999999;
  background-color: #cccccc;
  color: #666666;
}
.colorContent {
  color: red;
}
