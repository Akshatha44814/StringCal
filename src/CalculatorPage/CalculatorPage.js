import { useState } from "react";
import "./CalculatorPage.css";

const CalculatorPage = () => {
  const [result, setResult] = useState("");
  const [calculate, setCalculate] = useState("");
  const [error, setError] = useState(false);
  const [negativeNumbers, setNegativeNumbers] = useState([]);

  const inputMethod = (e) => {
    const inputValue = e.target.value.split("");
    if (inputValue.includes("-")) {
      setError(true);
      const res = e.target.value.match(/-?\d+/g).filter((a, b) => {
        if (a.includes("-")) {
          return true;
        }
      });
      setNegativeNumbers(res.join(","));
    } else {
      setError(false);
      const findNumber =
        inputValue.length > 0 &&
        inputValue.filter((a, b) => {
          if (!isNaN(parseInt(a))) {
            return true;
          }
        });
      const resultOutput =
        findNumber.length > 0
          ? findNumber?.reduce((c, d) => {
              return parseInt(c) + parseInt(d);
            })
          : 0;
      setCalculate(resultOutput);
    }
  };
  const calculateSum = () => {
    setResult(calculate);
  };
  return (
    <div className="conatinerPage">
      <div>
        <input
          type="text"
          className="input"
          placeholder="E.G.,//;\n1;2;3 or 1,2,3"
          onChange={(e) => inputMethod(e)}
        />
      </div>

      <button
        className="calculate"
        onClick={(e) => {
          calculateSum();
        }}
        disabled={error}
      >
        Calculate
      </button>
      {error ? (
        <span className="content colorContent">
          Negative numbers not allowed :{negativeNumbers}
        </span>
      ) : (
        <div className="resultInput">
          <input
            type="text"
            className="input"
            value={`${result}`}
            placeholder="Result Output"
            disabled
          />
        </div>
      )}
    </div>
  );
};
export default CalculatorPage;
