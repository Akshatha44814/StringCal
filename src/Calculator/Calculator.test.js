import { act, cleanup, fireEvent, render, screen } from "@testing-library/react";
import CalculatorPage from "../CalculatorPage/CalculatorPage";

afterEach(cleanup);
describe("renders Calculate Component", () => {
  it("Render Calculate Body Component", () => {
    render(<CalculatorPage />);
  });
  it("Input string Onclick calculate show output", () => {
    render(<CalculatorPage />);
    const inputStringField = screen.getByPlaceholderText(
      "E.G.,//;\\n1;2;3 or 1,2,3"
    );
    expect(inputStringField).toBeInTheDocument();
    const value = "1,2,3";
    fireEvent.change(inputStringField, {
      target: {
        value,
      },
    });
    expect(inputStringField).toHaveValue("1,2,3");

    const calculateSum = jest.fn();
    const { getByText } = render(<button onClick={() => calculateSum()} />);
    const buttonEl = getByText("Calculate");
    act(() => fireEvent.click(buttonEl));
    expect(buttonEl).toHaveTextContent("Calculate");
    expect(buttonEl).toBeTruthy();
    expect(buttonEl).toBeInTheDocument();
  });
});
