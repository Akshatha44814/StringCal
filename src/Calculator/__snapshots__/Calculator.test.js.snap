// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Calculate Component should render the value of negative number and error message 1`] = `
<DocumentFragment>
  <span
    class="content colorContent"
  >
    Negative numbers not allowed :-1,-2
  </span>
</DocumentFragment>
`;
